src/pyflame.cc

```diff
@@ -139,24 +194,16 @@ int main(int argc, char **argv) {
         std::this_thread::sleep_for(interval);
         PtraceAttach(pid);
       }
-      if (idle) {
-        std::cout << "(idle) " << idle << "\n";
-      }
-      // process the frames
-      for (const auto &kv : buckets) {
-        if (kv.first.empty()) {
-          std::cerr << "uh oh\n";
+      if (!include_ts) {
+        if (0 != print_frames(idle, buckets)) {
           return 1;
         }
-        auto last = kv.first.rend();
-        last--;
-        for (auto it = kv.first.rbegin(); it != last; ++it) {
-          std::cout << *it << ";";
-        }
-        std::cout << *last << " " << kv.second << "\n";
       }
     } else {
       const unsigned long frame_addr = FirstFrameAddr(pid, tstate_addr);
+      if (include_ts) {
```

Comment:
Cant we also just use print_frames or print_frames_ts (depending on include_ts)? It seems like we're really close to getting all of the stack printing in one place. For the one shot stack you might have to make a fake buckets and stick that one stack in there, or whatever, but it sure we be a nice cleanup.